datafusion/common/src/scalar/mod.rs

```diff
@@ -3646,18 +3646,22 @@ fn fmt_list(arr: ArrayRef, f: &mut fmt::Formatter) -> fmt::Result {
     write!(f, "{value_formatter}")
 }
 
+/// writes a byte array to formatter. `[1, 2, 3]` ==> `"1,2,3"`
+fn fmt_binary(data: &[u8], f: &mut fmt::Formatter) -> fmt::Result {
+    let mut first = true;
+    for b in data {
```

Comment:
I think this is slightly more idiomatic written as follows:
```
fn fmt_binary(data: &[u8], f: &mut fmt::Formatter) -> fmt::Result {
    let mut data_iter = data.iter();
    if let Some(b) = data_iter.next() {
        write!(f, "{}", b)?;
    }
    for b in data_iter {
        write!(f, ",{b}")?;
    }
    Ok(())
}
```

Reply:
Or even better
```
fn fmt_binary(data: &[u8], f: &mut fmt::Formatter) -> fmt::Result {
    if let Some(b) = data.first() {
        write!(f, "{}", b)?;
    }
    for b in data.iter().skip(1) {
        write!(f, ",{b}")?;
    }
    Ok(())
}
```

Reply:
I came up with something slightly different that I still think is more idiomatic in  https://github.com/apache/datafusion/pull/12323/commits/99a57e77b0ed6641982af65d8a33ef9b9a73296f.

```rust
/// writes a byte array to formatter. `[1, 2, 3]` ==> `"1,2,3"`
fn fmt_binary(data: &[u8], f: &mut fmt::Formatter) -> fmt::Result {
    let mut iter = data.iter();
    if let Some(b) = iter.next() {
        write!(f, "{b}")?;
    }
    for b in iter {
        write!(f, ",{b}")?;
    }
    Ok(())
}
```